File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string str0=strs[0];
        string ans;
        for (int i = 0; i  < str0.length(); i++)
        {
            for (int j = 1; j  < strs.size(); j++)
            {
                if (strs[j][i] != str0[i]) return ans;
            }
            ans+=str0[i];
        }
        return ans;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        StringBuilder ans= new StringBuilder();

        Arrays.sort(strs);
        String s1=strs[0];
        String s2=strs[(strs.length-1)];
        int i=0;
        while(i< s1.length() && i<s2.length()){
            if(s1.charAt(i)!=s2.charAt(i)){
                break;
            }else{
                ans.append(s1.charAt(i));
            }
            i++;
        }

        return ans.toString();
    }
}